### PR TITLE
[COREAPI-355] request bind helper

### PR DESCRIPTION
**Pull request by heggies**
## Description

> Please add your task in ClickUp here.

Task: [COREAPI-355](https://app.clickup.com/t/3857515/COREAPI-355)

## What's new?

> Please add if you added new improvement
- request bind helper (json, urlencoded, form-data, multipart/form-data)
- minor refactor on `PublishLog()`

## Examples
- `Content-Type: application/json`
<p align="center"></p>

- `Content-Type: application/x-www-form-urlencoded`
<p align="center"></p>

- `Content-Type: multipart/form-data` - no file
<p align="center"></p>

- `Content-Type: multipart/form-data`
<p align="center"></p>

### CODE REVIEW

File: v1/rest/bind.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/gin-gonic/gin"
+)
+
+type File *multipart.FileHeader
+type Files []*multipart.FileHeader
+
+// BindJSON params
+// 	@ctx: *gin.Context
+// 	@v: interface{}
+// 	return error
```
Comment (ahtuz) [inline v1/rest/bind.go:20]: @heggies @christyyy-a jadi skrg kita ketentuan godoc kasih tab 1x kah utk params & return nya?

**Review by ahtuz — CHANGES_REQUESTED** [review]

File: v1/rest/bind.go (hunk 1 above)
Comment (heggies) [reply v1/rest/bind.go:20]: biar beda line aja sih 
image

kalo gak dikasih tab 
image

Comment (ahtuz) [reply v1/rest/bind.go:20]: iya, apa kita mulai sekarang pake tab gitu aja utk godoc?

File: v1/rest/rest.go [hunk 2]
```diff
@@ -265,8 +265,9 @@ func PublishLog(context *gin.Context, status int, payload interface{}, msg ...st
 		return nil
 	}
 
+	contentType := context.GetHeader("Content-Type")
 	var requestBodyInterface map[string]interface{}
-	if len(requestBody) > 1 {
+	if contentType == "application/json" && len(requestBody) > 1 {
```
Comment (yafetzz) [inline v1/rest/rest.go:270]:
```suggestion
	if contentType == "application/json" && len(requestBody) > 0 {
```

pak titip tolong ubah dong

**Review by yafetzz — CHANGES_REQUESTED** [review]

**Review by yafetzz — APPROVED** [review]

File: v1/rest/bind.go (hunk 1 above)
Comment (heggies) [reply v1/rest/bind.go:20]: dari aku si yes, bagaimana ci @christyyy-a

**Review by ahtuz — APPROVED** [review]
dah oke sih

File: v1/rest/bind.go (hunk 1 above)
Comment (christyyy-a) [reply v1/rest/bind.go:20]: ini sepertinya sudah di bahas di clickup, oke kalo dari saya

**Review by christyyy-a — APPROVED** [review]

**Review by agusID — APPROVED** [review]

**Review by agusID — APPROVED** [review]

**Pull request merged**